sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/patterns.scala

```diff
@@ -63,7 +63,7 @@ object PhysicalOperation extends PredicateHelper {
         val substitutedFields = fields.map(substitute(aliases)).asInstanceOf[Seq[NamedExpression]]
         (Some(substitutedFields), filters, other, collectAliases(substitutedFields))
 
-      case Filter(condition, child) if condition.deterministic =>
+      case Filter(condition, child) if condition.deterministic || condition.isInstanceOf[And] =>
```

Comment:
I think this pattern should not return non-determinisitc exprs, but this current change does so, right?
Can we modify code to extract deterministic exprs in line 67-69?

Reply:
Yes, you are right. I will do it.

Reply:
It has been updated. Please review it, thanks~~